utils.go

```diff
@@ -47,7 +47,7 @@ func chooseProtocol(preferredIPProtocol string, target string, registry *prometh
 		}
 	}
 
-	probeDNSLookupTimeSeconds.Add(time.Since(resolveStart).Seconds())
+	defer probeDNSLookupTimeSeconds.Add(time.Since(resolveStart).Seconds())
```

Comment:
You still don't reach this line if the above return happens, this needs to move up.